pandas/tests/tseries/test_timezones.py

```diff
@@ -1228,6 +1228,27 @@ def f():
         dt = Timestamp('2013-11-03 01:59:59.999999-0400', tz='US/Eastern')
         assert dt.tz_localize(None) == dt.replace(tzinfo=None)
 
+    def test_replace_across_dst(self):
+        # GH#18319 check that 1) timezone is correctly normalized and
+        # 2) that hour is not incorrectly changed by this normalization
+        tz = pytz.timezone('US/Eastern')
+
+        ts_naive = Timestamp('2017-12-03 16:03:30')
+        ts_aware = tz.localize(ts_naive)
+
+        # Preliminary sanity-check
+        assert ts_aware == ts_aware.tzinfo.normalize(ts_aware)
+
```

Comment:
need to parametrize this across dateutil & pytz, so you need to move this to ``TestTimeZoneSupportPytz`` and use ``self.tz(...)``